tests/api_fastapi/core_api/routes/public/test_assets.py

```diff
@@ -0,0 +1,241 @@
+    assets = [
+        AssetModel(
+            id=i,
+            uri=f"s3://bucket/key/{i}",
+            extra={"foo": "bar"},
+            created_at=timezone.parse(default_time),
+            updated_at=timezone.parse(default_time),
+        )
+        for i in range(1, 1 + num)
+    ]
+    session.add_all(assets)
+    session.commit()
+
+
+def _create_provided_asset(session, asset: AssetModel) -> None:
+    session.add(asset)
+    session.commit()
+
+
+class TestAssets:
+    default_time = "2020-06-11T18:00:00+00:00"
+
+    @pytest.fixture(autouse=True)
+    def setup(self) -> None:
+        clear_db_assets()
+
+    def teardown_method(self) -> None:
+        clear_db_assets()
+
+    @provide_session
+    def create_assets(self, session, num: int = 2):
+        _create_assets(session=session, num=num)
+
+    @provide_session
+    def create_provided_asset(self, session, asset: AssetModel):
+        _create_provided_asset(session=session, asset=asset)
+
+
+class TestGetAssets(TestAssets):
+    def test_should_respond_200(self, test_client, session):
+        self.create_assets()
+        assets = session.query(AssetModel).all()
+        assert len(assets) == 2
+
+        response = test_client.get("/public/assets")
+        assert response.status_code == 200
+        response_data = response.json()
+        tz_datetime_format = self.default_time.replace("+00:00", "Z")
+        assert response_data == {
+            "assets": [
+                {
+                    "id": 1,
+                    "uri": "s3://bucket/key/1",
+                    "extra": {"foo": "bar"},
+                    "created_at": tz_datetime_format,
+                    "updated_at": tz_datetime_format,
+                    "consuming_dags": [],
+                    "producing_tasks": [],
+                    "aliases": [],
+                },
+                {
+                    "id": 2,
+                    "uri": "s3://bucket/key/2",
+                    "extra": {"foo": "bar"},
+                    "created_at": tz_datetime_format,
+                    "updated_at": tz_datetime_format,
+                    "consuming_dags": [],
+                    "producing_tasks": [],
+                    "aliases": [],
+                },
+            ],
+            "total_entries": 2,
+        }
+
+    def test_order_by_raises_400_for_invalid_attr(self, test_client, session):
+        response = test_client.get("/public/assets?order_by=fake")
+
+        assert response.status_code == 400
+        msg = "Ordering with 'fake' is disallowed or the attribute does not exist on the model"
+        assert response.json()["detail"] == msg
+
+    @pytest.mark.parametrize(
+        "url, expected_assets",
+        [
+            ("/public/assets?uri_pattern=s3", {"s3://folder/key"}),
+            ("/public/assets?uri_pattern=bucket", {"gcp://bucket/key", "wasb://some_asset_bucket_/key"}),
+            (
+                "/public/assets?uri_pattern=asset",
```

Comment:
Do not encode query parameters ourselves.

Use a dictionnary and leave the test_client to do that, passing the argument `params` to the call. (Also it's much more readable)